ec2/floatingip.py

```diff
@@ -0,0 +1,95 @@
+########
+# Copyright (c) 2014 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+# Boto Imports
+from boto.exception import EC2ResponseError
+from boto.exception import BotoServerError
+
+# Cloudify imports
+from cloudify import ctx
+from cloudify.exceptions import NonRecoverableError
+from cloudify.decorators import operation
+from ec2 import connection
+
+
+@operation
+def create(**kwargs):
+    """ Gets a floating IP from an Amazon Elastic IP.
+    """
+    ec2_client = connection.EC2ConnectionClient().client()
+    ctx.logger.debug('Getting Elastic IP from Amazon EC2.')
+
+    try:
+        address_object = ec2_client.allocate_address()
+    except (EC2ResponseError, BotoServerError) as e:
+        raise NonRecoverableError('(Node: {0}): Error. Failed to get '
+                                  'floating ip, returned: {1}.'
+                                  .format(ctx.instance.id, e))
+
+    ctx.logger.info('Elastic IP creation response: {0}'.format(address_object))
+    ctx.instance.runtime_properties['floatingip'] = address_object.public_ip
+
```

Comment:
can you log something like:

```python
ctx.logger.info('Elastic IP Provisioned: {0}'.format(address_object.public_ip1))
```

?

I think this is true for all logging throughout the plugin.
We should state that we're:

```python
ctx.logger.debug('Getting Elastic IP from Amazon EC2.')
```

then, if successful:

```python
ctx.logger.info('Elastic IP Provisioned: {0}'.format(address_object.public_ip))
```

Note the logging level. These are all 
or if failed:

```python
ctx.logger.info('Failed to provision Elastic IP')
```

note the logging level.